src/playercards/spawnbag/SpawnBag.ttslua

```diff
@@ -0,0 +1,202 @@
+require("playercards/PlayerCardSpawner")
+
+-- Allows spawning of defined lists of cards which will be created from the template in the All
+-- Player Cards bag.  SpawnBag.spawn will create objects based on a table definition, while
+-- SpawnBag.recall will clean them all up.  Recall will be limited to a small area around the
+-- spawned objects.  Objects moved out of this area will not be cleaned up.
+--
+-- SpawnSpec: Spawning requires a spawn specification with the following structure:
+-- {
+--   name: Name of this spawn content, used for internal tracking.  Multiple specs can be spawned,
+--       but each requires a separate name
+--   cards: A list of card IDs to be spawned
+--   globalPos: Where the spawned objects should be placed, in global coordinates.  This should be
+--       a valid Vector with x, y, and z defined, e.g. { x = 5, y = 1, z = 15 }
+--   rotation: Rotation for the spawned objects.  X=180 should be used for face down items.  As with
+--       globalPos, this should be a valid Vector with x, y, and z defined
+--   spread: Optional.  If present and true, cards will be spawned next to each other in a spread
+--       moving to the right.  globalPos will define the location of the first card, each after that
+--       will be moved a predefined distance
+-- }
+-- See BondedBag.ttslua for an example
+
+SpawnBag = { }
+
+-- To assist debugging, will draw a box around the recall zone when it's set up
+local SHOW_RECALL_ZONE = false
+
+local ALL_CARDS_GUID = "15bb07"
+
+-- Distance to expand the recall zone around any added object.
+local RECALL_BUFFER_X = 0.9
+local RECALL_BUFFER_Z = 0.5
+
+-- In order to mimic the behavior of the previous memory buttons we use a temporary bag when
+-- recalling objects.  This bag is tiny and transparent, and will be placed at the same location as
+-- this object.  Once all placed cards are recalled bag to this bag, it will be destroyed
+local RECALL_BAG = {
+  Name = "Bag",
+  Transform = {
+    scaleX = 0.01,
+    scaleY = 0.01,
+    scaleZ = 0.01,
+  },
+  ColorDiffuse = {
+    r = 0,
+    g = 0,
+    b = 0,
+    a = 0,
+  },
+  Locked = true,
+  Grid = true,
+  Snap = false,
+  Tooltip = false,
+}
+
+
+-- Tracks what has been placed by this "bag" so they can be recalled
+local placedSpecs = { }
+local placedObjectGuids = { }
+local recallZone = nil
+
+-- Loads a table of saved state, extracted during the parent object's onLoad
+SpawnBag.loadFromSave = function(saveTable)
+  placedSpecs = saveTable.placed
+  placedObjectGuids = saveTable.placedObjects
+  recallZone = saveTable.recall
+end
+
+-- Generates a table of save state that can be included in the parent object's onSave
+SpawnBag.getStateForSave = function()
+  local saveState = {
+    placed = placedSpecs,
+    placedObjects = placedObjectGuids,
+    recall = recallZone,
+  }
+
+  return saveState
```

Comment:
could get rid of line 76+77 and just replace the "local" in line 71 with "return"

Reply:
Done.